R/tm_a_pca.R

```diff
@@ -1,32 +1,37 @@
 #' Principal component analysis module
-#' @md
+#'
+#' Module conducts principal component analysis (PCA) on a given dataset and offers different
+#' ways of visualizing the outcomes, including elbow plot, circle plot, biplot, and eigenvector plot.
+#' Additionally, it enables dynamic customization of plot aesthetics, such as opacity, size, and
+#' font size, through UI inputs.
 #'
 #' @inheritParams teal::module
 #' @inheritParams shared_params
 #' @param dat (`data_extract_spec` or `list` of multiple `data_extract_spec`)
-#'   Columns used to compute PCA.
-#' @param alpha optional, (`numeric`) If scalar then the plot points will have a fixed opacity. If a
-#'   slider should be presented to adjust the plot point opacity dynamically then it can be a vector of
-#'   length three with `c(value, min, max)`.
-#' @param size optional, (`numeric`) If scalar then the plot point sizes will have a fixed size.
-#'   If a slider should be presented to adjust the plot point sizes dynamically then it can be a
-#'   vector of length three with `c(value, min, max)`.
-#' @param font_size optional, (`numeric`) font size control for title, x-axis label, y-axis label and legend.
-#'   If scalar then the font size will have a fixed size. If a slider should be presented to adjust the plot
-#'   point sizes dynamically then it can be a vector of length three with `c(value, min, max)`.
-#'
+#'   specifying columns used to compute PCA.
+#' @param alpha (`numeric`, optional) value or a vector of length three in the form `c(value, min, max)`.
```

Comment:
@insightsengineering/nest-core-dev We should align on how to write this "optional" indicator.

It seems we have quite a few usages on all of the teal packages _(even after all releases)_. Here are some that I've gathered from `{teal}`, `{teal.slice}` and `{teal.transform}`

1. ``@param optional (`character`) ...``
1. ``@param (optional `character`) ...``
1. ``@param (`character`, optional) ...``
1. ``@param (`character`) optional ...``

I've been using `2.` _(I think I saw that on the first `pre-release`)_ as it gives some flexibility when the parameter accepts multiple data types, but I think any of them is fine as long as we decide on a single one.

Reply:
If the type itself is not to be part of the following description, then (4) seems natural.

Reply:
I am okay with either of them
> @param optional (`character`) ...
 it could be mistaken as part of the parameter 

> @param (optional `character`) ...
> @param (`character`, optional) ...

my pick is adding "optional" before or after the data within parentheses as users often look for the type after/before its optional status.  its easy for the scan.

> `@param (character) optional ...`
it could be mistaken for part of description.

Reply:
I also prefer 4 and 1, over 2 and 3, so I'm fine with 4. @kartikeyakirar would you unify on your PRs in this package, and maybe you can create issues to get this covered in other packages? However I think we should start a discussion in a separate issue, so other team members are aware (in case they dont get notifications from review threads).

Reply:
Cool, let's do `4.` on this package and revisit the other on the near future

Reply:
I will  check which packages needs the changes and log issue for this. I will make changes for this in tmg as separate task  as `shared_param `and `ggplot2_args_multi ` also needs to be edited

Reply:
Or do you think this should warrant further discussion on its own issue? @m7pr @kartikeyakirar

Reply:
We could do `4.` for now _(for consistency sake and merge the PRs)_ and revert early next week with find+replace if we come to a different conclusion

Reply:
For `tmg` https://github.com/insightsengineering/teal.modules.general/issues/674

For the rest of the package https://github.com/insightsengineering/nestdevs-tasks/issues/52

Reply:
> Or do you think this should warrant further discussion on its own issue? @m7pr @kartikeyakirar

if needed we could continue this here https://github.com/insightsengineering/nestdevs-tasks/issues/52